org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Parser.java

```diff
@@ -13120,9 +13115,7 @@ public boolean visit(TypeDeclaration memberTypeDeclaration, ClassScope scope) {
 	}
 	boolean containsInitializers = false;
 	TypeDeclaration typeDeclaration = null;
-	for (int i = 0, max = result.length; i < max; i++) {
-		// parse each class or record body declaration
```

Comment:
apparently the refactoring is not safe wrt comments. 
If dogfooding should help improve the tool, than such flaws should be detected, reported and fixed, no?

Reply:
yes that's the purpose. 2 reviewers who did not see it. 6 eyes see more then 4 :-)